### PR TITLE
use div to wrap text section, not p

### PR DESCRIPTION
**Pull request by samhavens**
The text section was wrapped with p tags. If the HTML is complicated, such as a form, this makes it invalid. I replaced the p tags with divs and modified the query selector that was looking for the p tags to look for the div.

### CODE REVIEW

**Comment by PommeVerte** [comment]
+1 for this.

**Comment by t4t5** [comment]
I'm a bit reluctant on this one. SweetAlert is slowly going away from its general purpose of being an "alert"-replacement to being a general modal replacement.

I'm all for evolution though, we just need to rethink how to add custom HTML to the modal. Passing it as a string is simply too ugly. I'd rather come up with something better and refactor the library a little. :+1:

**Comment by samhavens** [comment]
@t4t5 Agreed. If you want to be generous, this is a stop-gap (i.e. it's a hack). I was trying to think of more elegant way to do this, but everything more structured felt less flexible.

If I can help, let me know.

**Comment by t4t5** [comment]
@samhavens What do you think of being able to add your sweetalert markup directly in the DOM? Like this:

```html
<div class="sweet-alert" id="delete-modal">
  <h2>Deleted!</h2>
  <p>Your file was deleted!</p>
  <button class="confirm">OK</button>
</div>
```

And then open if by passing a jQuery element or a DOM node to the swal-function:
```javascript
swal($('#delete-modal'));
```

I think it would solve a lot of problems that people have with "over-customizing", like adding a third button, implementing a form... etc.

**Comment by samhavens** [comment]
@t4t5 I think that's a good solution. Currently isn't there only one sweet
alert object, and it gets modified every time it pops up? Will having
multiples around throw anything off?

**Comment by t4t5** [comment]
@samhavens Yup, that's the problem we need to solve. ;) Maybe add an `original`-class to the current sweetAlert or something to distinguish it from the rest.

**Comment by samhavens** [comment]
@t4t5 Maybe this: if you pass in a string to swal(), it's behavior is
unchanged. If you want swal() to function as a modal, you set a new
parameter, content, to the selector of the div you want to use in the swal
object? I think there could still be only one then.

**Comment by SkyzohKeyx** [comment]
@t4t5 I think that having the markup and the parameters (via the data-* attributes) of the modal in the html would be really cool.
Then to open an alert you would have to pass the alert id to the `swal()` function.

Example:
```html
<div class="sweet-alert" id="delete-modal"
  data-cancelButton="true">
  <h2>Deleted!</h2>
  <p>Your file was deleted!</p>
  <button class="confirm">OK</button>
</div>
```
```javascript
swal('#delete-modal');
```
Would show the modal by adding it the class "current-sweetalert-modal" (to avoid collision with others current classes that the user could have in it's `CSS`.

**Comment by AkshatM** [comment]
I'm averse to the idea of introducing a special `div` in your code that defines the contents of your sweetalert in your HTML. Among other things, it'll:

- Mix style and control flow, which should be entirely separate. If I'm using Javascript, I should be able to accomplish what I want *in Javascript*. SweetAlert's main draw is that, at the moment, it functions like a pure JS solution - mixing with HTML is to me an ugly way to proceed, and detracts from its appeal.

- Requires a lot of manual effort. If I have a lot of SweetAlert messages, that's a lot of cruft code I'll have to write manually or generate somehow - neither of which make for clean injections. 

In my opinion, a much better solution would be to do it like React.js does: allow user-defined **types** and **templates** in Javascript for the SweetAlert message. You already have support for types, except that they are very limited - you have `input`,`success`,`error` and `warning`, each of which has its own look and feel. Here, each user-defined type would let you employ a template - HTML/CSS snippets with some blank or filler spaces you can fill in using a JS object. 

As a rough example of how it might work, the new types and templates might do a number of things:
- Let you define *keys* that correspond to certain values in your template. e.g. if we have a template defined by `<input type =  {this.input_type}>`, then one could do `swal('title': 'Hello', type='user-defined', input_type = 'file')` and have the SweetAlert render with an input box configured to accept files in place of text, with all other customisations intact. (This is also a great way to avoid XSS attacks, since you can't arbitrarily inject HTML anymore!)
- Let you define how to *handle* your input. For instance, in the context of this issue, a user could specify that all text be handled as `<div>{this.text}</div>` instead of `<p>{this.text}</p>` as is currently being done.
- Let you define templates for *different sections* of a sweet alert. You could define a template for the title, a template for buttons, etc.
- Let you define different *actions* e.g. instead of just slide-up or slide-down, somebody could define their own action/transition for the SweetAlert.

The bottom line is that you can achieve a pure JS solution with support against XSS and more powerful customisation tools without resorting to HTML and additional data attributes as a crutch. 

What do you think?

**Comment by samhavens** [comment]
I was thinking about the XSS problem today. This is clearly a solution to that, and goes well beyond.

I think from the issues it's clear that sweetalert is being used as a modal, so it may as well be done right. I like this idea.

**Comment by AkshatM** [comment]
We could recommend/use a [client-side templating engine](http://garann.github.io/template-chooser/) to structure it all. I personally favour React.js because it supports type validation out of the box when you pass in properties to a component, but of course anything else could work. 

I think the only problem with this solution might be that this would require a significant rewrite. Specifically, `sweetalert-dev.js` has a `switch` statement with an object case that simply updates the default params, rather than let you define your own. Like @t4t5 mentions, there is only one SweetAlert box that keeps getting updated each time. We need a way to 'subclass' objects - to have a base `SweetAlert` object that simply checks the `type` argument supplied, and then calls the template involved. 

One way to do this - and the one I think would be most painless, given how extensive SweetAlert is - would be to have a function called `registerTemplate()`, which registers a template in a store accessible to `swal`. `swal` would then check the `type` argument on being invoked against an internal list -`success`,`input`,`error`,`warning` - and, if not found, default to passing all other arguments to the template and somehow render it. I think `_sweetAlertInitialize$getModal$getOverlay$getInput$setFocusStyle$openModal$resetInput$fixVerticalPosition` might be able to handle the rendering, although I'd have to take a closer look at it, I'm not sure.

**Comment by t4t5** [comment]
This can now be achieved with the new [content option](https://sweetalert.js.org/docs/#content) in SweetAlert 2.0.
